console/network/src/lib.rs

```diff
@@ -194,6 +194,7 @@ pub trait Network:
     const MAX_OUTPUTS: usize = 16;
 
     /// The maximum program depth.
+    /// This restriction is **not** in use.
```

Comment:
IMO you can remove.

Reply:
Should `MAX_PROGRAM_DEPTH` be removed entirely?

Reply:
For clarity, `MAX_PROGRAM_DEPTH` is no longer relevant because upgradability can introduce import cycles. However, we are still bounded on the function level with MAX_TRANSITIONS. Is that correct?

If it is, then we can remove `MAX_PROGRAM_DEPTH` as long as we can guarantee there is no deployment rejected because of this rule.

Reply:
That's correct, and to clarify do you mean no existing deployment rejected?

Reply:
I meant to check if that any deployments were rejected because of the `MAX_PROGRAM_DEPTH`. Because if the rule is removed then block processing may be different.

Reply:
I just double-checked, deployments are rejected in 2 cases:
- Deployment already exists
- Failed `finalize_deployment`

Reply:
Ah gotcha. `Stack::initialize`, which is called during `verify_deployment` will current check `MAX_PROGRAM_DEPTH`. This would be a "verification aborted" transaction which is not a "rejected" transaction so we do not need to create a migration. Is that correct?

Reply:
I believe that is correct. Would appreciate a double check here to make sure that removing this condition does not affect existing onchain state.

Reply:
A genesis to tip sync test should be sufficient right?

Reply:
A genesis to tip sync will be run before the next release.